### ISSUE TITLE
Keep Players in group upon death, Group member removal menu

### ISSUE THREAD

**Issue by AttackingJoe** [AttackingJoe/The-Company-Mission-Framework#17]
I would like to have players remain in their orignal group up death, and adding either a add action or ace menu to the group leader to remove the player. Thus creating a need to confirm a player is killed.

**Comment by Smillmeister** [comment]
This is likely to be extremely hard to program arma bins players names 2 seconds after death holding then in memory would require scripting that I have previously tried to achieve with no success.

**Comment by AttackingJoe** [comment]
I don't think we need to store their name? simply readd them to their orginal group when they respawn.

**Comment by thealmightyn00b** [comment]
The problem with this is the way the spec script works is it deletes everything about the player when he dies. ST Hud references this to populate itself. The only solution is to set up some sort of database, be it just a bunch of global variables or something like extDB. Both of those require a substantial amount of work that i just don't think is worth it. Now if you want to decompile ST HUD and add a function to not remove people you go right ahead, but the frame work is not the place to fix this.

**Comment by AttackingJoe** [comment]
Why is the spec script deleting all of the information relevant to the player upon their death? shouldn't it just respawn then and execute the camera script

**Comment by thealmightyn00b** [comment]
The spec script isnt deleting shit, ArmA itself is.

**Comment by Smillmeister** [comment]
Like I and noob said arma bins the info on player death or 2 seconds thereafter therefore you would have to retain the players name in memory after death in order to "remove it" manually from the group. It's annoying as fuck but the reality is passing and holding players names in memory is fiddly as fuck

**Comment by AttackingJoe** [comment]
That isn't actually correct, the information doesn't vanish, it is simply removed from the players former body and placed into his new one in specpen. When a unit re-spawns using the base spawn they are kept in their group. I've found a section of the spec script that assigns the player to a null group, My guess is removing this bit of code will give the desired effect. I will verify this after work today.

**Comment by Smillmeister** [comment]
The player inhabits a vehicle which is the character model, when you shift them to specpen the move vehicles to a new character model the information is simply no longer available to assign how you want to assign it. It still exists but what now exists is a new instance of the player.

**Comment by AttackingJoe** [comment]
Yes, but somewhere in that information it says that this player is part of
this group, until the spec script reassigns their group.

**Comment by Smillmeister** [comment]
Be my guest to try I pissed away two weeks trying to hold or recall things from memory. He'll if you get it to work great! :)

**Comment by AttackingJoe** [comment]
I don't think you and I were trying to accomplish the same things though. You where trying to call info off an object (IE you interact with this corpse and it tells you who it was, however that information has moved on to the next player body). What i'm doing will essentially be calling the info that's in the new body already.


(I also vote we call the new info "spirit" from now on)

**Comment by AttackingJoe** [comment]
Worked as intended
https://www.youtube.com/watch?v=WQuamS2blYw&feature=youtu.be

Now just need a UI to kick player from group
